### ISSUE TITLE
No snapshot from cameras

### ISSUE THREAD

**Issue by eskehagen** [AdyRock/com.onvif#41]
Hi, 
I have tried to get snapshots from my Nedis SmartLife IP cameras in my Homey Pro, but with no succes.
I have read your other issues on this site, but cannot find a solution to solve my problems. 

The cameras are added to the Homey, and accesseble in the app. But it's not possible to get a snapshot or move detection from the camera. 
I'm not sure if the settings in the ONVIF app is correct for the camera or if I miss something ?

I have added a detailed log for the case: adding the new camera to the app. 
[onvif app log.txt.txt](https://github.com/AdyRock/com.onvif/files/10346957/onvif.app.log.txt.txt)

**Comment by AdyRock** [comment]
The first issue shown is:
19:34:40.143: Get camera capabilities error (192.168.68.122): ONVIF SOAP Fault
To gather more information, could you change the Log Level option to Full Information and then restart the app. Then go back and send the log again.

The image issue is shown here:
```
19:34:40.602: Snapshot URL: http://www.gwelltimes.com/manage/Upload/Pic/20147914514701.jpg

19:34:40.725: Fetching (192.168.68.122) Motion Event image with no Auth from: http://www.gwelltimes.com/manage/Upload/Pic/20147914514701.jpg
19:34:41.433: SnapShot error (192.168.68.122): request to http://www.gwelltimes.com/manage/Upload/Pic/20147914514701.jpg failed, reason: socket hang up
19:34:41.438: Fetching (192.168.68.122) Motion Event image with Basic Auth. From: http://www.gwelltimes.com/manage/Upload/Pic/20147914514701.jpg
19:34:41.627: SnapShot error (192.168.68.122): request to http://www.gwelltimes.com/manage/Upload/Pic/20147914514701.jpg failed, reason: socket hang up
19:34:41.641: Fetching (192.168.68.122) Motion Event image with Digest Auth. From: http://www.gwelltimes.com/manage/Upload/Pic/20147914514701.jpg
19:34:41.706: SnapShot error (192.168.68.122): request to http://www.gwelltimes.com/manage/Upload/Pic/20147914514701.jpg failed, reason: socket hang up
19:34:41.717: Fetch MOTION error (192.168.68.122): {
  "ok": false,
  "statusText": "ECONNRESET"
}
19:34:41.722: Event SnapShot error (192.168.68.122): ECONNRESET
```

So the link provided by the camera for the image is not responding. Maybe you can contact the camera manufacture and ask about this. Normally the image is stored locally on the camera but this one seems to be in the cloud.

**Comment by eskehagen** [comment]
Thanks for quick answer. 

I have restarted the app with Full Info log enabled. 
It's attached here: 
[FullInfoLog_Nedis.txt](https://github.com/AdyRock/com.onvif/files/10360766/FullInfoLog_Nedis.txt)

I have been in contact with Nedis and got this answer: 
_Unfortunately we do not support third party software that use the ONVIF signal from the camera.
Please contact Homey support for this.
 
The snapshots in our app are basically screenshots made of the videoclip that is recorded.
This is a function of out app/server, not necessarily of the camera.
How the Homey software processes this we do not know.
Perhaps you can make it record through Homey itself to get that working?_ 

I'm not sure if it's possible to record the camera view through Homey - or what to do now ?

**Comment by AdyRock** [comment]
Unfortunately Homey doesn't have the processing power to decode the live video stream. 
Can you ask them if it's possible to read the images from the server?

**Comment by eskehagen** [comment]
I have been in contact with Nedis again, but no success.
They said it's not possible to stream from the camera from a website. But they didn't answer directly, when I asked how the Onvif works on the Nedis camera. And sounds like they didn't know how to get it working.

Do you have more thinks to try?
I'm a bit lost.

**Comment by AdyRock** [comment]
All I can see is they somehow make an image on the server that is not accessible.
Maybe they create a different file each time and remove the old one then update the ONVIF setting, but the app currently caches the URL. 
I will have a look at providing an option to fetch the URL just before fetching the image to see if that helps.

**Comment by BrianBrinkmann** [comment]
Hi Adrian

I have a similar problem with my Arenti cameras
No motion alert received and cannot get the snapshot to work.

I did write to Arenti support to find some help.
Their response was this:

Dear Brian

I apologize for the inconvenience, the P2Q camera supports ONVIF functionality.
The device connects to NVR/DVR via onvif on port 8000  
APP Click to enter the live video page - Settings - Advanced Functions - ONVIF
After turning onvif function on, you can use RTSP to access the camera stream.

URL:
rtsp://admin:password@IP:8554/Streaming/Channels/101
rtsp://admin:password@IP:8554/Streaming/Channels/102

Note: 101 is the streaming address of the main stream and 102 is the streaming address of the sub-stream.

admin: onvif username
password: onvif password
IP: IP of the camera
8554: port
________________________________________

Best Regards 
Arenti & Laxihub Support Team 

I am no expert in ONVIF nor cameras.
Have tried to change several settings but with no luck.

basically I am just looking for motion alerts and if possible a chance to push the image via a message in Homey.
Also changing the motion setting to ON when the home is empty 

can you provide some kind of setup guide?
I have been looking but did not find it

thanks 
Brian

**Comment by AdyRock** [comment]
Glad you had a response. 
One thing to note is Homey won't support the rtsp stream, it can only read a snapshot (e.g. jpg).
If you can open the app settings page and select the Diagnostics tab, set the Diagnostics log to Detailed and then restart the app. Once it has started, go back to the Log and send it so I can see what features it reports.

**Comment by BrianBrinkmann** [comment]
Hi Adrian

Thanks - the log was sent just before.

When I click the camera icon, I always get ECONNRESET - Se screen shot
Also none of the values seem to update - see screen shot

Regards Brian


tor. 25. jan. 2024 kl. 10.04 skrev Adrian Rockall ***@***.***

**Comment by AdyRock** [comment]
Everything looks OK in the log. Could you point your web browser to http://192.168.2.8:8000/snapshot to find out if you can see an image?
If you have to login then try the username and password that you specified in to connect the camera to the app. 
If you don't get anything then ask Arenti why. Tell them that ONVIF is reporting that URL for the snapshot. 

I can also see the pull request is being registered correctly, so you should get MOTION notifications. If you clear the log, then move in front of the camera, you should see notifications arrive.

**Comment by BrianBrinkmann** [comment]
Hi,

Tested but does not work with motion events.
Also the snapshot URL does not work either

Arenti suppprt says they do not support snapshot.

Seems like they say ONVIF is supported - but not really….

Mvh Brian


tor. 25. jan. 2024 kl. 10.28 skrev Adrian Rockall ***@***.***

**Comment by BrianBrinkmann** [comment]
Response from Arenti support :

Dear Brian

After confirming with the R&D team, this is because the feature is not yet
perfected and the snapshot feature is not currently supported but will be
available when the feature is perfected.

I was told by the R&D team that it would take some more time to complete
this feature, so I can't confirm exactly when.

________________________________________


Best Regards

Arenti & Laxihub Support Team

Cathy

But the motion push should be working - I guess - maybe that is also not
perfected yet😱

Mvh Brian


tor. 25. jan. 2024 kl. 10.48 skrev Brian Brinkmann <
***@***.***>:
